docs/src/FPMPNLPModel.md

```diff
@@ -0,0 +1,192 @@
+  `OFList`|`Vector{H}` | List of largest representable numbers corresponding to `FPList`
+  `γfunc`| | dot product error model function callback.
+  `ωfRelErr`|`Vector{H}` | List of relative error factor for objective function evaluation corresponding to FP formats in `FPList`
+  `ωgRelErr`|`Vector{H}` |List of relative error factor for gradient evaluation corresponding to FP formats in `FPList`
+  `ObjEvalMode` |`Int` | objective function error evaluation mode
+  `GradEvalMode` |`Int` | gradient error evaluation mode
+  `X`| `B` | Container used for interval evaluation, memory pre-allocation
+  `G`| `B` | Container used for interval evaluation, memory pre-allocation
+
+## `γfunc`
+
+ `γfunc` callback function provides the error on dot product: $|x.y - fl(x.y)|\leq |x|.|y|\gamma func(n,u)$ with $n$ the dimension of $x$ and $y$ vector and $u$ the unit-roundoff of the FP format used to perform the dot product operation.
+ The expected template is: `γfunc(n::Int,u::AbstractFloat)` with `n` the dimension of the problem and `u` the unit roundoff of the considered FPFormat.
+ By default, `γfunc(n,u) = n*u` is used.
+This function is used to take finite-precision norm computation error into account, to further guarantee gradient error bounds (details below).
+
+# Constructors
+
+1. `FPMPNLPModel(Model::AbstractNLPModel{D,S},FPList::Vector{K}; kwargs...) where {D,S,K<:DataType}`
+
+2. `FPMPNLPModel(f,x0, FPList::Vector{DataType}; kwargs...)`
+Build a `ADNLPModel` from `f` and `x0` and call constructor 1.
+
+## Keyword arguments
+  + `HPFormat=Float64` : high precision format (must be at least as accurate as `FPList[end]`), corrensponds to `H` parameter after instanciation
+  + `γfunc=nothing` : use default callback if not provided (see Fields section above)
+  + `ωfRelErr=HPFormat.(sqrt.(eps.(FPList)))`: use relative error model by default for objective evaluation
+  + `ωgRelErr=HPFormat.(sqrt.(eps.(FPList)))`: use relative error model by default for gradient evaluation
+  + `obj_int_eval = false` : if true, use interval arithmetic for objective value and error evaluation
+  + `grad_int_eval = false` : if true, use interval arithmetic for gradient value and error evaluation
+
+## Checks upon instanciation
+Some checks are performed upon instanciation. These checks include:
++ Length consistency of vector fields:  `FPList`, `EpsList`, `UList`
++ `HPFormat` is at least as accurate as the highest precision floating point format in `FPList`. Ideally HPFormat is more accurate to ensure the numerical stability.
++ Interval evaluations: it might happen that interval evaluation of objective function and/or gradient is type-unstable or returns an error. The constructor returns an error in this case. This type of error is most likely due to [IntervalArithmetic.jl](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/master/README.md) package.
++ `FPList` is ordered by increasing floating point format accuracy
```

Comment:
```suggestion
+ Length consistency of vector fields:  `FPList`, `EpsList`, `UList`;
+ `HPFormat` is at least as accurate as the highest precision floating point format in `FPList`. Ideally HPFormat is more accurate to ensure the numerical stability;
+ Interval evaluations: it might happen that interval evaluation of objective function and/or gradient is type-unstable or returns an error. The constructor returns an error in this case. This type of error is most likely due to [IntervalArithmetic.jl](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/master/README.md) package;
+ `FPList` is ordered by increasing floating point format accuracy.
```